### PR TITLE
[ADD] ir_cron: allow env variable to force cron_trigger db notification

### PR DESCRIPTION
**Pull request by sts-odoo**
--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by sts-odoo** [comment]
@antonylesuisse

**Review by Julien00859 — APPROVED** [review]
LGTM, just you can use `os.getenv` instead `os.environ.get`, a few keystroke less to type!

**Comment by sts-odoo** [comment]
@odoo/rd-security

File: odoo/addons/base/models/ir_cron.py [hunk 1]
```diff
@@ -447,13 +452,16 @@ def _trigger_list(self, at_list):
...
 
     def _notifydb(self):
-        """ Wake up the cron workers """
+        """ Wake up the cron workers
+        The ODOO_NOTIFY_CRON_CHANGES environment variable allows to force the notifydb on both
+        ir_cron modification and on trigger creation (regardless of call_at)
+        """
         with odoo.sql_db.db_connect('postgres').cursor() as cr:
-            cr.execute('NOTIFY cron_trigger')
+            cr.execute('NOTIFY cron_trigger, %s', (self.env.cr.dbname,))
```
Comment (xmo-odoo) [inline odoo/addons/base/models/ir_cron.py:464]: You sneaky little sneak, that's what you really were after weren't you!

FWIW `cr.execute` can take a list of parameters and it's usually more readable / more reliable, especially when there's a single parameter (avoids the need for a trailing comma).

File: odoo/addons/base/models/ir_cron.py [hunk 2]
```diff
@@ -371,6 +374,8 @@ def _try_lock(self, lockfk=False):
 
     def write(self, vals):
         self._try_lock()
+        if 'nextcall' in vals and os.getenv('ODOO_NOTIFY_CRON_CHANGES'):
```
[the remark this answers is not in the document]
Comment (sts-odoo) [reply odoo/addons/base/models/ir_cron.py]: indeed

**Review by amigrave — APPROVED** [review]

**Comment by amigrave** [comment]
@robodoo r+

**Pull request closed without merging**